### PR TITLE
Updates

### PR DESCRIPTION
**Pull request by webseo-onilne**
Security Shutters
Shutters in markup manager
Updates for proposal system

### CODE REVIEW

File: admin/admin-html.php [hunk 1]
```diff
@@ -120,18 +90,40 @@
...
+					<td><input id="" ng-model="dimRestraints.maxDrop" class="dims-input range_1_markup_by" type="text" /></td>
+					<td><input id="" ng-model="dimRestraints.minDrop" class="dims-input range_1_markup_by" type="text" /></td>
+					<td><button class="markup-button button button-primary button-small" ng-disabled="!dimRestraints" ng-click="saveDims(dimRestraints)">Save</button></td>
+				</tr>
+			</tbody>
+		</table>		
+
+<!-- 		<div class="meta-details">
+		<h4 ng-cloak>Price Sheet Meta Details: <span class="sheet-name" ng-cloak>{{friendlyName}}</span></h4>
 			<p ng-cloak>Min-Width: <strong>{{dimRestraints.minWidth}}<small>mm</small></strong></p>
```
Comment (michaeldoye) [inline admin/admin-html.php]: Need to revert these changes, no longer require max/min dims to be edited

File: admin/bd-made-to-measure-admin.php [hunk 2]
```diff
@@ -30,6 +30,8 @@ public function __construct() {
 					add_action( 'wp_ajax_get_markup_ajax', array( &$this, 'get_markup_ajax' ) );
 					// Save markup
 					add_action( 'wp_ajax_bd_ajax_save_markup_data', array( &$this, 'bd_ajax_save_markup_data' ) );
+					// Save dims
+					add_action( 'wp_ajax_bd_ajax_save_dimension_data', array( &$this, 'bd_ajax_save_dimension_data' ) );					
 					// Get price books
```
Comment (michaeldoye) [inline admin/bd-made-to-measure-admin.php]: This can be reverted, no longer require max/min dims to be edited

File: admin/bd-made-to-measure-admin.php [hunk 3]
```diff
@@ -192,6 +194,13 @@ public function bd_plugin_settings() {
 				include plugin_dir_path( __FILE__ ) . "admin-settings.php";
 			}
 
+			/**
+			 * TODO - Plugin settings page
+			 */	
+			public function bd_dimension_restraints() {
+				include plugin_dir_path( __FILE__ ) . "bd-dim-restraints.php";
+			}			
+
```
Comment (michaeldoye) [inline admin/bd-made-to-measure-admin.php]: Revert, re: max/min dims

File: admin/admin-html.php (hunk 1 above)
Comment (webseo-onilne) [inline admin/admin-html.php:117]: Revert, re: max/mins dims, editing not required

File: admin/bd-made-to-measure-admin.php (hunk 2 above)
Comment (webseo-onilne) [inline admin/bd-made-to-measure-admin.php:35]: Revert, re: max/mins dims, editing not required

File: admin/bd-made-to-measure-admin.php (hunk 3 above)
Comment (webseo-onilne) [inline admin/bd-made-to-measure-admin.php:204]: - Revert, re: max/mins dims, editing not required.
- Delete unused file

File: admin/bd-made-to-measure-admin.php [hunk 4]
```diff
@@ -364,6 +399,32 @@ public function bd_ajax_save_markup_data() {
...
+						WHERE `price_group` = '$variation_name' ", OBJECT );
+					
+					// Send the reponse back
+					echo empty( $update_dims ) ? json_encode( $wpdb->last_error ) : json_encode( $results );
+
+				}
+
+				wp_die();				
+			}			
+
```
Comment (webseo-onilne) [inline admin/bd-made-to-measure-admin.php:428]: Revert, re: max/mins dims, editing not required

File: assets/js/admin/bd_admin_custom.js [hunk 5]
```diff
@@ -164,6 +166,29 @@ jQuery(document).ready(function ($) {
...
+          };
+
+          console.log(response);
+
+          toastr.success('Dimensions Added', 'Success');
+          NProgress.done();          
+        });
+      }
+
       function addMarkup(data) {
```
Comment (webseo-onilne) [inline assets/js/admin/bd_admin_custom.js:192]: Revert, re: max/mins dims, editing not required

File: assets/js/frontend/blinds.js [hunk 6]
```diff
@@ -4,33 +4,83 @@ app
...
+        jQuery('.notice').html(blinds.max_drop_error + ': '+parseInt(bd_dims.max_drop)).css('color', 'red');
+        jQuery('.single_variation_wrap, .final-price').fadeOut();
+        return;
+
+      } 
+      else if (drop < parseInt(bd_dims.min_drop)) {
+        jQuery('.notice').html(blinds.min_drop_error + ': '+parseInt(bd_dims.min_drop)).css('color', 'red');
+        jQuery('.single_variation_wrap, .final-price').fadeOut();
+        return;
+
```
Comment (webseo-onilne) [inline assets/js/frontend/blinds.js:49]: Need to sort this jQuery out - replace with angular.element or move to directive

**Comment by webseo-onilne** [comment]
Cannot Commit, please review comments and make required changes

**Comment by webseo-onilne** [comment]
Cannot Commit, please review comments and make required changes
